### PR TITLE
Added Visual Studio features section.

### PR DESCRIPTION
**Pull request by markodoseninfinum**

### CODE REVIEW

File: Education/Visual Studio features.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+- `ctril+s` - to save a file.
+- `ctrl+e,d` - to format a file.
+- `ctrl+w` - to close a file (needs to be mapped in 'Keyboard' menu to command 'File.Close').
+- `ctrl+t` - to navigate through files, properties, methods etc.
+- `ctrl+g` - to navigate to specific line number in the file. 
+- `ctrl+f` - to search through files.
+- `ctrl+h` - open find and replace menu. 
+- `alt+enter` or `ctrl+.`- to open intelisense for the line you are at.
+- `ctrl+F12` - to navigate to method implementation or class definition.
```
Comment (icottiero) [inline Education/Visual Studio features.md:64]: Here I would add differentiation between `F12` and `ctrl+F12`
- `F12` - navigate to the code definition
- `ctrl+F12` - navigate to the implementation or open list of implementations if there are more than one

File: Education/Visual Studio features.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+- `ctrl+r,a` - to run all tests.
+- `ctrl+d` - to make a copy of the line you are at.
+- `F2` - to rename a file, property or method.
+
```
Comment (icottiero) [inline Education/Visual Studio features.md:68]: Could we also add shortcuts (these are the ones that I use quite often):
- `ctrl+shift+L` - delete the current line
- `shitf+del` - cut the current line
- `ctrl+š,ctrl+s` - navigate to the current file in solution explorer

File: Education/Visual Studio features.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+Different projects require different setting configurations, like connection string to database or URL to external API etc. These configurations in .NET are usually stored in appsettings.json file. Visual studio allows you to create different configurations through 'Configuration Manager' found in the 'Build' tab. By default projects are run in 'Debug' configuration with appsettings.json but it can be changed to use different settings, example appsettings.Test.json.
+
+Additionally VS allows you to use 'user secrets' by right clicking on project and clicking on 'Manage User Secrets'. These secrets are stored on your machine and cannot be pushed to github. These secrets override appsettings.json and are useful for local development.
+
+
+
+### Connecting to Azure 
+
+VS can be connected to Azure account by clicking on 'Manage connections' in 'Team' tab. When VS is connected to Azure it allows you a several different features, such as accessing DevOps git repository, selecting specific existing trigger (like blob or queue) from your Azure account etc. These Azure connections can also be used to connect to Azure database or other resources and manage them through VS itself.
```
Comment (mapernar) [inline Education/Visual Studio features.md:46]: Maybe change part `When VS is connected to Azure it allows you a several different features` into `When VS is connected to Azure it allows you to use several different features`

File: Education/Visual Studio features.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+
+
+### Shortcuts
+
+VS provides shortcuts to some most common actions.  All shortcuts can be found by clicking on 'Options' under 'Tools' tab in the 'Keyboard' menu. Note that some extensions modify or add new shortcuts. Shortcuts can also be exported or imported.
+
+Some of the most common that we use:
+
+- `ctril+s` - to save a file.
```
Comment (mapernar) [inline Education/Visual Studio features.md:56]: `ctrl` instead of `ctril`

File: Education/Visual Studio features.md (hunk 2 above)
Comment (mapernar) [inline Education/Visual Studio features.md:68]: Maybe add `ctrl+q - global search`

**Review by mapernar — CHANGES_REQUESTED** [review]

File: Education/Visual Studio features.md (hunk 2 above)
Comment (dominfinum) [reply Education/Visual Studio features.md:68]: You can also set up tracking current file by default, so VS will automatically do that for you for each file you switch to.  Tools -> Options -> Projects and Solutions -> General

Comment (dominfinum) [inline Education/Visual Studio features.md:68]: We can also add some of these shortcuts:

> ctrl+k+c - comment selected block of code
> ctrl+k+u - uncomment selected block of code
> ctrl+m+o - collapse all methods in current file
> ctrl+m+p - expand all methods in current file
> ctrl+- - back
> ctrl+shift+f - find in files
> crtl+shift+h - replace in files

File: Education/Visual Studio features.md (hunk 1 above)
Comment (markodoseninfinum) [reply Education/Visual Studio features.md:66]: Added.

File: Education/Visual Studio features.md (hunk 2 above)
Comment (markodoseninfinum) [reply Education/Visual Studio features.md:76]: Added.

File: Education/Visual Studio features.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+- Fine Code Coverage - provides an overview of your test coverage giving information such as percentage of code covered, risky hotspot, covered branches etc.
+- Format document on Save - automatically applies default code formatting when saving a document.
+- Output enhancer - provides a better visual graphics when checking IDE console logs.
+- Visual Studio Spell Checker - adds spell checking for all files.
+
+
+
+### App settings configuration
+
+Different projects require different setting configurations, like connection string to database or URL to external API etc. These configurations in .NET are usually stored in appsettings.json file. Visual studio allows you to create different configurations through 'Configuration Manager' found in the 'Build' tab. By default projects are run in 'Debug' configuration with appsettings.json but it can be changed to use different settings, example appsettings.Test.json.
```
[the remark this answers is not in the document]
Comment (markodoseninfinum) [reply Education/Visual Studio features.md]: Fixed.

File: Education/Visual Studio features.md (hunk 3 above)
Comment (markodoseninfinum) [reply Education/Visual Studio features.md]: Fixed.

File: Education/Visual Studio features.md (hunk 2 above)
Comment (markodoseninfinum) [reply Education/Visual Studio features.md:76]: Added.

Comment (markodoseninfinum) [reply Education/Visual Studio features.md:76]: Added.

File: Education/Visual Studio features.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+Some of these templates provide additional configuration in the setup itself so the code generated is immediately ready to use. For example, Azure Functions project allows you to chose what type of trigger do you want for your function. It also allows you to setup a connection string to the trigger so the generated code can be run and debug locally.
+
+Visual Studio also allows you to create your own templates for the project. The instructions on how to create your custom template can be found on this [link](https://docs.microsoft.com/en-us/visualstudio/ide/how-to-create-project-templates?view=vs-2022).
+
+
+
+### Extensions
+
+VS allows you to add extensions that enhance or alter functionalities provided by the IDE. There are a lot extensions that can be found at 'Extensions' tab or searched on the internet. Some of the extensions are free and some of them are paid.
```
[the remark this answers is not in the document]
Comment (markodoseninfinum) [reply Education/Visual Studio features.md]: Fixed.

**Comment by markodoseninfinum** [comment]
@mapernar can you review the changes again?

**Review by mapernar — APPROVED** [review]

**Pull request merged**